executor/insert_common.go

```diff
@@ -508,16 +511,36 @@ func (e *InsertValues) getRowInPlace(ctx context.Context, vals []types.Datum, ro
 
 // getColDefaultValue gets the column default value.
 func (e *InsertValues) getColDefaultValue(idx int, col *table.Column) (d types.Datum, err error) {
-	if !col.DefaultIsExpr && e.colDefaultVals != nil && e.colDefaultVals[idx].valid {
-		return e.colDefaultVals[idx].val, nil
+	if e.colDefaultVals != nil && e.colDefaultVals[idx].valid {
+		if !col.DefaultIsExpr {
+			return e.colDefaultVals[idx].val, nil
+		}
+		return table.EvalColDefaultExpr(e.ctx, col.ToInfo(), e.colDefaultVals[idx].expr)
+	}
+	var (
+		defaultVal  types.Datum
+		defaultExpr ast.ExprNode
+	)
+	if !col.DefaultIsExpr {
+		defaultVal, err = table.GetColDefaultValue(e.ctx, col.ToInfo())
+	} else {
+		defaultExpr, err = table.GetColDefaultExpr(col.ToInfo())
```

Comment:
How about cache the default expression in `TableCommon`?

Reply:
it's practicable, but i think it's ok here